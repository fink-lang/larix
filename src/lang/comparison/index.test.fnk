{describe, it, expect, to_match_snapshot} = import '@fink/jest'

{parse_expr} = import '../../'


describe 'comparison', fn:

  it 'parses ==', fn:
    expect
      parse_expr 'a == b'
      to_match_snapshot

    expect
      parse_expr 'a + b == c'
      to_match_snapshot

    expect
      parse_expr 'a == b == c'
      to_match_snapshot


  it 'parses !=', fn:
    expect
      parse_expr 'a != b'
      to_match_snapshot

    expect
      parse_expr 'a != b != c'
      to_match_snapshot

    expect
      parse_expr 'a + b != c'
      to_match_snapshot


  it 'parses <=', fn:
    expect
      parse_expr 'a <= b'
      to_match_snapshot

    expect
      parse_expr 'a <= b <= c'
      to_match_snapshot

    expect
      parse_expr 'a + b <= c'
      to_match_snapshot


  it 'parses >=', fn:
    expect
      parse_expr 'a >= b'
      to_match_snapshot

    expect
      parse_expr 'a >= b >= c'
      to_match_snapshot

    expect
      parse_expr 'a + b >= c'
      to_match_snapshot


  it 'parses >', fn:
    expect
      parse_expr 'a > b'
      to_match_snapshot

    expect
      parse_expr 'a > b > c'
      to_match_snapshot

    expect
      parse_expr 'a + b > c'
      to_match_snapshot


  it 'parses <', fn:
    expect
      parse_expr 'a < b'
      to_match_snapshot

    expect
      parse_expr 'a < b < c'
      to_match_snapshot

    expect
      parse_expr 'a + b < c'
      to_match_snapshot

    expect
      parse_expr
        '
          a + b <incomplete_jsx
        '
      to_match_snapshot

