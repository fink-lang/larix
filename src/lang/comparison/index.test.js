import {parse_expr} from '../../';


describe('comparison', ()=> {

  it('parses single line: a == b', ()=> {
    expect(
      parse_expr(`a == b`)
    ).toMatchSnapshot();
  });

  it('parses single line: a != b', ()=> {
    expect(
      parse_expr(`a != b`)
    ).toMatchSnapshot();
  });


  it('parses single line: a <= b', ()=> {
    expect(
      parse_expr(`a <= b`)
    ).toMatchSnapshot();
  });


  it('parses single line: a >= b', ()=> {
    expect(
      parse_expr(`a >= b`)
    ).toMatchSnapshot();
  });


  it('parses single line: a > b', ()=> {
    expect(
      parse_expr(`a > b`)
    ).toMatchSnapshot();
  });


  it('parses single line: a < b', ()=> {
    expect(
      parse_expr(`a < b`)
    ).toMatchSnapshot();
  });
});
