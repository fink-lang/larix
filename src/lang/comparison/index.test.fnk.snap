// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comparison parses != 1`] = `
comp != (1:0-1:6)
  ident (1:0-1:1) a
  ident (1:5-1:6) b
`;

exports[`comparison parses != 2`] = `
comp != (1:0-1:11)
  comp != (1:0-1:6)
    ident (1:0-1:1) a
    ident (1:5-1:6) b
  ident (1:10-1:11) c
`;

exports[`comparison parses != 3`] = `
comp != (1:0-1:10)
  arithm + (1:0-1:5)
    ident (1:0-1:1) a
    ident (1:4-1:5) b
  ident (1:9-1:10) c
`;

exports[`comparison parses < 1`] = `
comp < (1:0-1:5)
  ident (1:0-1:1) a
  ident (1:4-1:5) b
`;

exports[`comparison parses < 2`] = `
comp < (1:0-1:9)
  comp < (1:0-1:5)
    ident (1:0-1:1) a
    ident (1:4-1:5) b
  ident (1:8-1:9) c
`;

exports[`comparison parses < 3`] = `
comp < (1:0-1:9)
  arithm + (1:0-1:5)
    ident (1:0-1:1) a
    ident (1:4-1:5) b
  ident (1:8-1:9) c
`;

exports[`comparison parses < 4`] = `
arithm + (1:0-3:0)
  ident (1:0-1:1) a
  call (1:4-3:0)
    ident (1:4-1:5) b
    jsx:elem (1:6-2:1) c

error:
undefined:3:0
1| a + b <
2| c
3| 
   ^

Expected one of [ '/>' ] but found undefined.



`;

exports[`comparison parses <= 1`] = `
comp <= (1:0-1:6)
  ident (1:0-1:1) a
  ident (1:5-1:6) b
`;

exports[`comparison parses <= 2`] = `
comp <= (1:0-1:11)
  comp <= (1:0-1:6)
    ident (1:0-1:1) a
    ident (1:5-1:6) b
  ident (1:10-1:11) c
`;

exports[`comparison parses <= 3`] = `
comp <= (1:0-1:10)
  arithm + (1:0-1:5)
    ident (1:0-1:1) a
    ident (1:4-1:5) b
  ident (1:9-1:10) c
`;

exports[`comparison parses == 1`] = `
comp == (1:0-1:6)
  ident (1:0-1:1) a
  ident (1:5-1:6) b
`;

exports[`comparison parses == 2`] = `
comp == (1:0-1:10)
  arithm + (1:0-1:5)
    ident (1:0-1:1) a
    ident (1:4-1:5) b
  ident (1:9-1:10) c
`;

exports[`comparison parses == 3`] = `
comp == (1:0-1:11)
  comp == (1:0-1:6)
    ident (1:0-1:1) a
    ident (1:5-1:6) b
  ident (1:10-1:11) c
`;

exports[`comparison parses > 1`] = `
comp > (1:0-1:5)
  ident (1:0-1:1) a
  ident (1:4-1:5) b
`;

exports[`comparison parses > 2`] = `
comp > (1:0-1:9)
  comp > (1:0-1:5)
    ident (1:0-1:1) a
    ident (1:4-1:5) b
  ident (1:8-1:9) c
`;

exports[`comparison parses > 3`] = `
comp > (1:0-1:9)
  arithm + (1:0-1:5)
    ident (1:0-1:1) a
    ident (1:4-1:5) b
  ident (1:8-1:9) c
`;

exports[`comparison parses >= 1`] = `
comp >= (1:0-1:6)
  ident (1:0-1:1) a
  ident (1:5-1:6) b
`;

exports[`comparison parses >= 2`] = `
comp >= (1:0-1:11)
  comp >= (1:0-1:6)
    ident (1:0-1:1) a
    ident (1:5-1:6) b
  ident (1:10-1:11) c
`;

exports[`comparison parses >= 3`] = `
comp >= (1:0-1:10)
  arithm + (1:0-1:5)
    ident (1:0-1:1) a
    ident (1:4-1:5) b
  ident (1:9-1:10) c
`;
